CASC/PortfolioMode.cpp

```diff
@@ -143,12 +144,16 @@ bool PortfolioMode::searchForProof()
     //we normalize now so that we don't have to do it in every child Vampire
     ScopedLet<Statistics::ExecutionPhase> phaseLet(env.statistics->phase,Statistics::NORMALIZATION);
     
-    if (env.options->shuffleInput()) { // instead to "combing things into shape" we shuffle and ruffle them
-      Shuffling().shuffle(*_prb);
-    } else {
+    if (env.options->normalize()) { // set explicitly by CASC(SAT) and SMTCOMP modes
       Normalisation().normalise(*_prb);
     }
 
+    // note that this is shuffleInput for the master process (for exceptional/experimental use)
+    // the usual way is to have strategies request shuffling explicitly in the schedule strings
+    if (env.options->shuffleInput()) {
```

Comment:
Out of interest, what's the interaction between normalization and shuffling... I guess that if we normalize before shuffling we'll get something different than if we don't normalize? But, in general, it seems a waste of work to normalize then shuffle (?)

Reply:
A good shuffle should make any previous normalization a waste of effort. A good normalization should undo any shuffle. 

However, unfortunately, our Normalization wasn't perfect last time I looked and I didn't find a way to fix it (cheaply). It boils down to graph isomorphism problem again, so somewhere down the parse tree it must decided, e.g., that "a" > "b" or vice versa. But that might be a detail compared to the fact we cannot "normalize" the signature order, so ---most glaringly---`sp=occurrence` will always refer to the occurrence in the original (perhaps Geoff-shuffled) input, not in the normalized one.